Dshop/apps/products_catalogue/tests/test_api_persistent_cart.py

```diff
@@ -0,0 +1,148 @@
+        assert Decimal(data_item['subtotal'])
+        total += Decimal(product.price) * quantity
+        count += quantity
+    assert Decimal(data['total']) == total
+    assert data['count'] == count
+
+
+def assert_data_empty(data):
+    assert data['total'] == '0.00'
+    assert data['count'] == 0
+    assert data['items'] == []
+
+
+@pytest.mark.django_db
+def test_get_cart_empty(api_client):
+    response = api_client.get(reverse("api_cart"), {}, format="json")
+    assert response.status_code == status.HTTP_200_OK
+    assert_data_empty(response.data)
+
+
+@pytest.mark.django_db
+def test_add(api_client, tv_product):
+    data = {
+        'items': [ {'product_pk': tv_product.pk, 'quantity': 10} ] 
+    }
+    response = api_client.post(reverse("api_cart"), data, format='json')
+    assert_products_data(response.data, [tv_product], [10])
+    assert response.status_code == status.HTTP_201_CREATED
+    response = api_client.get(reverse("api_cart"), {}, format='json')
+    assert response.status_code == status.HTTP_200_OK
+    assert_products_data(response.data, [tv_product], [10])
+
+
+@pytest.mark.django_db
+def test_add_relogin_get(tv_product):
+    client = APIClient()
+    user = User.objects.create_user(username='testuser', password='testpassword')
+    client.force_authenticate(user)
+    data = {
+        'items': [ {'product_pk': tv_product.pk, 'quantity': 10} ] 
+    }
+    response = client.post(reverse("api_cart"), data, format='json')
+    assert response.status_code == status.HTTP_201_CREATED
+    assert_products_data(response.data, [tv_product], [10])
+    client.force_authenticate(user=None)
+    client.force_authenticate(user)
+    
+    response = client.get(reverse("api_cart"), {}, format='json')
+    assert response.status_code == status.HTTP_200_OK
+    assert_products_data(response.data, [tv_product], [10])
+    
+
+@pytest.mark.django_db
+def test_add_ten_and_get(api_client, ten_tv_products):
+    quantities = [1, 1, 6, 8, 3, 4, 2, 26, 1, 10]
+
+    data = {   
+    'items': [
+            {'product_pk': product.pk, 'quantity': quantity}
+            for product, quantity in zip(ten_tv_products, quantities)
+        ]
+    }
+    response = api_client.post(reverse("api_cart"), data, format="json")
+    assert response.status_code == status.HTTP_201_CREATED
+    assert_products_data(response.data, ten_tv_products, quantities)
+    response = api_client.get(reverse("api_cart"), {}, format="json")
+    assert response.status_code == status.HTTP_200_OK
+    assert_products_data(response.data, ten_tv_products, quantities)
+
+
+@pytest.mark.django_db
+def test_delete_ten(api_client, ten_tv_products):
+    quantities = [1, 1, 6, 8, 3, 4, 2, 26, 1, 10]
+    data = {   
+    'items': [
+            {'product_pk': product.pk, 'quantity': quantity}
+            for product, quantity in zip(ten_tv_products, quantities)
+        ]
+    }
+    response = api_client.post(reverse("api_cart"), data, format="json")
```

Comment:
One last thing for all tests
please remove format=json.  It's not needed. 
https://github.com/pymasterspl/Dshop/blob/e9bfd0e3087063920873fffc251e0b33304fb04c/Dshop/Dshop/settings.py#L214

Reply:
I got used to another project without this option set. Will fix it ...